source/features/render-semantic-types.tsx

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import {$$} from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager.js';
+
+function isSemanticCommitTitleElement(commitTitle: string): boolean {
+	// This may not be fully accurate.
+	// Tests, if the title string
+	// starts with a word,
+	// followed by an anything in parentheses,
+	// followed by an optional exclamation mark,
+	// followed by a colon,
+	// followed by anything.
+	return /^\w*(?:\(.*\))?!?:.*/.test(commitTitle);
+}
+
+const defaultCommitTypes = new Set([
+	'feat',
+	'fix',
+	'chore',
+	'docs',
+	'build',
+	'refactor',
+	'test',
+	'ci',
+	'perf',
+]);
+
+function isSemanticCommitTitleTypeDefault(semanticCommitTitle: string): boolean {
+	// This may not be fully accurate.
+	// Gets the first word from the title.
+	const type = /^(\w*)/.exec(semanticCommitTitle)?.pop();
+
+	if (!type) {
+		return false;
+	}
+
+	return defaultCommitTypes.has(type);
+}
+
+function extractSemanticCommitTitleTypeAndScope(semanticCommitTitle: string): [string, string?] {
+	// Gets the first word from the title.
+	const type = /^(\w*)/.exec(semanticCommitTitle)!.pop()!;
+
+	// Gets the word in parentheses from the title.
+	const scope = /\((\w*)\)/.exec(semanticCommitTitle)?.pop();
+
+	return [type, scope];
+}
```

Comment:
These two functions should be a single regex.exec call for detection and extraction, as suggested above.